packages/expo-modules-core/android/src/main/java/expo/modules/interfaces/taskManager/TaskServiceProviderHelper.kt

```diff
@@ -0,0 +1,36 @@
+package expo.modules.interfaces.taskManager
+
+import android.content.Context
+import expo.modules.core.ModulePriorities
+import expo.modules.core.interfaces.Package
+
+class TaskServiceProviderHelper {
+  companion object {
+    /*
+      Uses reflection to look through the current list of packages and attempts to find
+      one that provides a TaskServiceInterface implementation.
+     */
+    fun getTaskServiceImpl(context: Context): TaskServiceInterface? {
+      // Use reflection to get the packages list from ExpoModulesPackageList without
+      // creating the reactInstanceManager:
+      val expoModules: Class<*>? = try {
+        Class.forName("expo.modules.ExpoModulesPackageList")
+      } catch (e: ClassNotFoundException) {
+        // Handle the exception, e.g., log it or fallback to a default behavior
+        return null
+      }
+      val getPackageList = expoModules?.getMethod("getPackageList") ?: return null
```

Comment:
let's add a note to `getPackageList` informing readers in that place that if they change something there, they might break this?

Reply:
Good point - the `ExpoModulesPackageList` is generated by auto linking this pattern is already used in the same way in two other places in `expo-modules-core` so I thought it was safe without any additional explanation.

Reply:
It still might make sense to add it [here](https://github.com/expo/expo/blob/724049acb1ef56e0ed4645ada080d1c524b000d8/packages/expo-modules-autolinking/src/platforms/android.ts#L115)

but ... I guess making changes there would be fairly major change and this would get caught.

Reply:
And this will also get caught [here](https://github.com/expo/expo/blob/main/packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/ExpoModulesHelper.kt) and [here](https://github.com/expo/expo/blob/main/packages/expo/android/src/main/java/expo/modules/ExpoModulesPackage.kt) as well :)